astropile/build_master_cat.py

```diff
@@ -0,0 +1,106 @@
+        in the master catalogue for that dataset.
+    matching_radius : float, optional
+        The maximum separation between two sources in the catalogues to be
+        considered a match, by default 1.0 [arcsec].
+
+    Returns
+    -------
+    master_cat : pd.DataFrame
+        The master catalogue containing the combined information from all the
+        input catalogues.
+    """
+
+    # Set the columns for the master catalogue
+    master_cat = pd.DataFrame(
+        columns=["ra", "dec", "healpix"] + names + [f"{name}_idx" for name in names]
+    )
+
+    for cat, name in zip(cats, names):
+        # Extract the relevant columns
+        cat = extract_cat_params(cat)
+
+        # Match the catalogues
+        master_coords = SkyCoord(master_cat.loc[:, "ra"], master_cat.loc[:, "dec"], unit="deg")
+        cat_coords = SkyCoord(cat.loc[:, "ra"], cat.loc[:, "dec"], unit="deg")
+        idx, sep2d, _ = master_coords.match_to_catalog_sky(cat_coords)
+        mask = sep2d < matching_radius * units.arcsec
+
+        # Update the matching columns
+        master_cat.loc[mask, name] = True
+        master_cat.loc[mask, name + "_idx"] = idx[mask]
+
+        # Add new rows to the master catalogue
+        if len(master_cat) == 0:
+            # keep everything for first catalogue
+            mask = np.zeros(len(cat), dtype=bool)
+        else:
+            # match to master catalogue so far
+            idx, sep2d, _ = cat_coords.match_to_catalog_sky(master_coords)
+            mask = sep2d < matching_radius * units.arcsec
+        idx = np.arange(len(cat), dtype=int)
+        name_data = []
+        name_idx_data = []
+        for subname in names:
+            if subname != name:
+                # Add rows for each catalogue. These are False becaue they didn't match
+                name_data.append(np.zeros(np.sum(~mask), dtype=bool))
+                name_idx_data.append(-np.ones(np.sum(~mask), dtype=int))
+            else:
+                # Add rows for the current catalogue. These are True because they are the current catalogue
+                name_data.append(np.ones(np.sum(~mask), dtype=bool))
+                name_idx_data.append(idx[~mask])
+        # Collect the new rows into a DataFrame
+        append_cat = pd.DataFrame(
+            columns=["ra", "dec", "healpix"] + names + [f"{name}_idx" for name in names],
+            data=np.stack(
+                [cat.loc[~mask, col] for col in ["ra", "dec", "healpix"]]
+                + name_data
+                + name_idx_data
+            ).T,
+        )
+
+        # Append the new rows to the master catalogue
+        master_cat = pd.concat([master_cat, append_cat], ignore_index=True)
+
+    # Convert the columns to the correct data types
+    master_cat["ra"] = master_cat["ra"].astype(float)
+    master_cat["dec"] = master_cat["dec"].astype(float)
+    master_cat["healpix"] = master_cat["healpix"].astype(int)
+    for name in names:
```

Comment:
correct me if I'm wrong, but you already specify the bool and int type above (see lines 79-80 and 83-84, 84 you might need to specify dtype=int), probably don't need to do it again right?

Reply:
Also, you can probably specify the dtype for ra dec and healpix above as well and get rid of this whole code chunk?